packages/x-tree-view/src/SimpleTreeView/SimpleTreeView.test.tsx

```diff
@@ -128,98 +118,67 @@ describe('<SimpleTreeView />', () => {
   });
 
   it('should call onKeyDown when a key is pressed', () => {
-    const handleKeyDown = spy();
+    const handleTreeViewKeyDown = spy();
+    const handleTreeItemKeyDown = spy();
 
-    const { getByRole } = render(
-      <SimpleTreeView onKeyDown={handleKeyDown}>
-        <TreeItem nodeId="test" label="test" data-testid="test" />
+    const { getByTestId } = render(
+      <SimpleTreeView onKeyDown={handleTreeViewKeyDown}>
+        <TreeItem nodeId="one" data-testid="one" onKeyDown={handleTreeItemKeyDown} />
       </SimpleTreeView>,
     );
     act(() => {
-      getByRole('tree').focus();
+      getByTestId('one').focus();
```

Comment:
Relates a lot to #12428

> Nitpick: It seems possible to avoid the usage of testId by keeping the getByRole.
We could add the explicit "name" field if we so desire extra stability.

It's mostly for consistency across the tests, the vast majority uses `testId`.
I think we can align on a single approach in #12428 and use it everywhere while migrating all the tests to `describeTreeView`.

> Nitpick2: Is there a specific reason why we are using the methods de-structured from the render function instead of the root import of screen?

`screen` is the advised way by the created or `react-testing-library` from what I remember, so no specific reason.
This topic will go away with #12428 if we create the right abstractions (which I think we should for readability).

Reply:
> It's mostly for consistency across the tests, the vast majority uses `testId`.

`getByRole` has ~2x more usages than `getByTestId` (and almost all of them are on TreeView tests). 🤔 
IMHO, it would be best to use `getByTestId` when there is no other way to select elements. 🙈

Reply:
![image](https://github.com/mui/mui-x/assets/3309670/bf95476c-4732-4fe4-a452-983ce47e126f)
![image](https://github.com/mui/mui-x/assets/3309670/bc57f350-3703-4c97-95b1-7f9400fa2f6f)

:thinking: not sure I have the same results :laughing:

Reply:
(and all `getByRole` are for `tree` AFAIK)

Reply:
Oh, I was referring to the global usage in the `mui-x` codebase, sorry for the confusion. 🙈